src/Engine/ProtoAssociative/CodeGen.cs

```diff
@@ -264,10 +264,10 @@ private GraphNode GetFinanlStatementOfSSA(GraphNode subNode, ref int index)
         // http://en.wikipedia.org/wiki/Tarjan's_strongly_connected_components_algorithm
         // 
         // Returns the first strongly connected component
-        private List<GraphNode> StrongConnectComponent(GraphNode node, ref int index, Dictionary<GraphNode, int> lowlinkMap, Dictionary<GraphNode, int> indexMap, Stack<GraphNode> S)
+        private List<GraphNode> StrongConnectComponent(GraphNode node, ref int index, Dictionary<int, int> lowlinkMap, int[] indexList, Stack<GraphNode> S)
         {
-            indexMap[node] = index;
-            lowlinkMap[node] = index;
+            indexList[node.dependencyGraphListID] = index;
+            lowlinkMap[node.dependencyGraphListID] = index;
```

Comment:
I can see how the original `GraphNode` type could be used as a unique key, however, I'm not so sure if the `dependencyGraphListID` will be unique enough; in other words, that it may be overwritten.